webapp/components/ContentMenu/ContentMenu.spec.js

```diff
@@ -0,0 +1,463 @@
+import { config, mount, createLocalVue } from '@vue/test-utils'
+import Vuex from 'vuex'
+import VTooltip from 'v-tooltip'
+import Styleguide from '@human-connection/styleguide'
+import ContentMenu from './ContentMenu.vue'
+
+const localVue = createLocalVue()
+
+localVue.use(Styleguide)
+localVue.use(VTooltip)
+localVue.use(Vuex)
+
+config.stubs['router-link'] = '<span><slot /></span>'
+
+let getters, mutations, mocks, menuToggle, openModalSpy
+
+describe('ContentMenu.vue', () => {
+  beforeEach(() => {
+    mocks = {
+      $t: jest.fn(str => str),
+      $i18n: {
+        locale: () => 'en',
+      },
+      $router: {
+        resolve: jest.fn(obj => {
+          obj.href = '/post/edit/d23a4265-f5f7-4e17-9f86-85f714b4b9f8'
+          return obj
+        }),
+        push: jest.fn(),
+      },
+    }
+  })
+
+  describe('mount', () => {
+    mutations = {
+      'modal/SET_OPEN': jest.fn(),
+    }
+    getters = {
+      'auth/isModerator': () => false,
+      'auth/isAdmin': () => false,
+    }
+
+    const openContentMenu = (values = {}) => {
+      const store = new Vuex.Store({ mutations, getters })
+      const wrapper = mount(ContentMenu, {
+        propsData: {
+          ...values,
+        },
+        mocks,
+        store,
+        localVue,
+      })
+      menuToggle = wrapper.find('.content-menu-trigger')
+      menuToggle.trigger('click')
+      return wrapper
+    }
+
+    describe('owner of contribution', () => {
+      let wrapper
+      beforeEach(() => {
+        wrapper = openContentMenu({
+          isOwner: true,
+          resourceType: 'contribution',
+          resource: {
+            id: 'd23a4265-f5f7-4e17-9f86-85f714b4b9f8',
+          },
+        })
+        openModalSpy = jest.spyOn(wrapper.vm, 'openModal')
+      })
+
+      it('can edit the contribution', () => {
+        expect(
+          wrapper
+            .findAll('.ds-menu-item')
+            .filter(item => item.text() === 'post.menu.edit')
+            .at(0)
+            .find('span.ds-menu-item-link')
+            .attributes('to'),
+        ).toBe('/post/edit/d23a4265-f5f7-4e17-9f86-85f714b4b9f8')
+      })
+
+      it('can delete the contribution', () => {
+        wrapper
+          .findAll('.ds-menu-item')
+          .filter(item => item.text() === 'post.menu.delete')
+          .at(0)
+          .trigger('click')
+        expect(openModalSpy).toHaveBeenCalledWith('delete')
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-11-11T21:58:04Z" title="Monday, November 11th 2019, 10:58:04 pm +01:00">Nov 11, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,543 @@
+import { config, mount, createLocalVue } from '@vue/test-utils'
+import Vuex from 'vuex'
+import VTooltip from 'v-tooltip'
+import Styleguide from '@human-connection/styleguide'
+import ContentMenu from './ContentMenu.vue'
+
+const localVue = createLocalVue()
+
+localVue.use(Styleguide)
+localVue.use(VTooltip)
+localVue.use(Vuex)
+
+config.stubs['router-link'] = '<span><slot /></span>'
+
+let getters, mutations, mocks, menuToggle, openModalSpy
+
+describe('ContentMenu.vue', () => {
+  beforeEach(() => {
+    mocks = {
+      $t: jest.fn(str => str),
+      $i18n: {
+        locale: () => 'en',
+      },
+      $router: {
+        resolve: jest.fn(obj => {
+          obj.href = '/post/edit/d23a4265-f5f7-4e17-9f86-85f714b4b9f8'
+          return obj
+        }),
+        push: jest.fn(),
+      },
+    }
+  })
+
+  describe('mount', () => {
+    mutations = {
+      'modal/SET_OPEN': () => jest.fn(),
+    }
+    getters = {
+      'auth/isModerator': () => false,
+      'auth/isAdmin': () => false,
+    }
+
+    const openContentMenu = (values = {}) => {
+      const store = new Vuex.Store({ mutations, getters })
+      const wrapper = mount(ContentMenu, {
+        propsData: {
+          ...values,
+        },
+        mocks,
+        store,
+        localVue,
+      })
+      menuToggle = wrapper.find('.content-menu-trigger')
+      menuToggle.trigger('click')
+      return wrapper
+    }
+
+    describe('contribution', () => {
+      it('owner can edit and delete contribution', () => {
+        const wrapper = openContentMenu({
+          isOwner: true,
+          resourceType: 'contribution',
+          resource: {
+            id: 'd23a4265-f5f7-4e17-9f86-85f714b4b9f8',
+          },
+        })
+        let items = wrapper.findAll('.ds-menu-item')
+        openModalSpy = jest.spyOn(wrapper.vm, 'openModal')
+        expect(items).toHaveLength(2)
+        expect(items.at(0).text()).toBe('post.menu.edit')
+        expect(
+          items
+            .at(0)
+            .find('span.ds-menu-item-link')
+            .attributes('to'),
+        ).toBe('/post/edit/d23a4265-f5f7-4e17-9f86-85f714b4b9f8')
+        expect(items.at(1).text()).toBe('post.menu.delete')
+        expect(
+          items
+            .at(1)
+            .find('span.ds-menu-item-link')
+            .attributes('to'),
+        ).toBe('/post-menu-delete')
+        items.at(1).trigger('click')
+        expect(openModalSpy).toHaveBeenCalledWith('delete')
```


As we talked about briefly on `Discord` @Mogge, please try to avoid multiple expectations in the same it block... sometimes, it's kinda unavoidable, but in this case, I think it's not.

you can add a `beforeEach` inside `describe contribution`, which handles similar set up, then have one expectation per it block, describing what you are testing with the `expect`